### ISSUE TITLE
Parallel execution of SPOT

### ISSUE THREAD

**Issue by c-mertes** [BennyStrobes/SPOT#1]
Dear @BennyStrobes,

thank you for providing the codebase for SPOT. We currently run it as is on some GTEx tissues, but were wondering if there is any option to parallelize the execution? It loops through every gene/cluster in a single thread, which takes for some of the tissues up to 10 days on a Intel(R) Core(TM) i7-6700 CPU @ 3.40GHz for roughly 20k clusters per tissue. 

As the tests on each cluster are independent, is there a nice fix to parallelize it? Any hint or solution would be appreciated. 

Best,
Christian

**Comment by BennyStrobes** [comment]
Hi Christian,

Apologies for the delay. The SPOT packages does not have any parallelization built in, currently. But you are correct that each cluster is independent, from the perspective of SPOT. To parallelize in the current the implementation, just split the input file into smaller, equal sized chunks. I hope this helps.

Ben

**Comment by c-mertes** [comment]
Yes, parallelizing on the chunked input files is what we currently ended up with. But I think it would be nice to have this native in spot. As we basically divided the runtime by the number of cores with the overhead of compiling the STAN model.
